app/src/main/java/com/x8bit/bitwarden/ui/platform/components/button/BitwardenFilledIconButton.kt

```diff
@@ -1,37 +1,39 @@
-package com.x8bit.bitwarden.ui.platform.components.icon
+package com.x8bit.bitwarden.ui.platform.components.button
 
+import androidx.annotation.DrawableRes
 import androidx.compose.material3.FilledIconButton
 import androidx.compose.material3.Icon
-import androidx.compose.material3.IconButtonColors
+import androidx.compose.material3.IconButtonDefaults
 import androidx.compose.material3.MaterialTheme
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.semantics.semantics
 import androidx.compose.ui.tooling.preview.Preview
 import com.x8bit.bitwarden.R
-import com.x8bit.bitwarden.ui.platform.components.model.IconResource
 import com.x8bit.bitwarden.ui.platform.components.util.rememberVectorPainter
 import com.x8bit.bitwarden.ui.platform.theme.BitwardenTheme
 
 /**
- * An icon button that displays an icon from the provided [IconResource].
+ * A filled icon button that displays an icon.
  *
- * @param iconRes Icon to display on the button.
+ * @param vectorIconRes Icon to display on the button.
+ * @param contentDescription The content description for this icon button.
  * @param onClick Callback for when the icon button is clicked.
- * @param isEnabled Whether or not the button should be enabled.
  * @param modifier A [Modifier] for the composable.
+ * @param isEnabled Whether or not the button should be enabled.
  */
 @Composable
-fun BitwardenIconButtonWithResource(
-    iconRes: IconResource,
+fun BitwardenFilledIconButton(
+    @DrawableRes vectorIconRes: Int,
+    contentDescription: String,
```

Comment:
`Icon` itself tends to allow this to be nullable:

```kotlin
    contentDescription: String?,
```

Do we definitely need this be required? I suppose maybe because its a button?

Reply:
I wanted to make sure we enforced that a content description was passed in. A button should always announce _something_.

Reply:
Yeah, make sense!